rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/service/SpringBeanService.java

```diff
@@ -21,4 +26,22 @@ public SpringBeanService(GraphContext context)
     {
         return super.findAllByProperty(SpringBeanModel.SPRING_BEAN_NAME, name);
     }
+
+    /**
+     *
+     * @param application
+     * @return an unmodifiable list of SpringBeanModel entries for the given application
+     */
+    public List<SpringBeanModel> findAllByApplication(ProjectModel application)
+    {
+        List<SpringBeanModel> models = new ArrayList<>();
+        for (SpringBeanModel model : findAll())
+        {
+            if (application.equals(model.getApplication()))
+            {
+                models.add(model);
+            }
```

Comment:
It would be better to filter at the DB query level. Imagine getting 1 000 000 spring beans from 1000 apps. That means 1000 x running through 1 000 000 beans filtering out a fraction of them. Not that I expect 1mil beans, but we should pay attention to performance.